### DISCUSSION TITLE
Request for Urgent Assistance Regarding Error in Termination of MEC Applications in Simu5G

### DISCUSSION THREAD

**Discussion by rimsayegh** [Unipisa/Simu5G#238] [Q&A]
I am currently working on Simu5G and have encountered a strange issue. As with any MEC application, we typically call the terminateMECApp function from the VIM to stop it. For an MEC app that requires a service, we also need to disconnect the internal MEPlatform gates from the MEService gates to properly terminate the app.

The issue arises when we call terminateMECApp. It seems that it cannot access the locationService to disconnect it from the MEC app, as the serviceIndex remains at -2 throughout the simulation. I should note that the location service is available on the MEC host and works as expected, but the VIM does not have access to the location service module, and the MEC services vector in the VIM is empty.

I have attached the relevant parts of the code along with the error message for your reference.

I would greatly appreciate your urgent assistance in resolving this issue.
<img width="473" alt="terminate2" src="https://github.com/user-attachments/assets/7d03498b-8085-4a02-987e-fe1981f6b2eb">
<img width="431" alt="terminate1" src="https://github.com/user-attachments/assets/8e592000-ba29-4368-9121-fef1593a1ae0">
<img width="437" alt="FindService" src="https://github.com/user-attachments/assets/38039acc-d066-47a8-af82-264aadb016ff">
<img width="638" alt="error" src="https://github.com/user-attachments/assets/44355897-e3d4-4676-8977-960dde4d8cb2">

**Comment by giovanninardini** [comment]
Are you sure that the problem is the one you mention? 
Even if the serviceIndex is -2, then the code does not throw an error. Actually, if you use the LocationService as shown in the "singleMecHost" example the serviceIndex is not used at all.

**Comment by linofex** [comment]
Hi, 
the error refers to something related to the "processedHttpMessage". Maybe it has been already deleted and the descructor does not check it before. It should be investigated.. 

Moreover, **now** the connection between Services and MEC apps occures via TCP and not via direct gates.
The code you posted is obsolete and not used anymore by the MEC services. It was written for an old version.

**Comment by rimsayegh** [comment]
Thank you so much for taking the time to help me with the issue I posted. 
Your explanation was clear, and it helped me understand where I was going wrong.
